src/requiredBy.js

```diff
@@ -0,0 +1,12 @@
+export default function requiredBy(requiredByPropName, propType, defaultValue = null) {
+  return function validator(props, propName, componentName, ...rest) {
+    if (props[requiredByPropName]) {
```

Comment:
ah right, so `requiredByPropName` is assumed to be a boolean, or rather, it's assumed that "truthy" means "it's there" and "falsy" means "it's not"?